source/termination_criteria/end_walltime.cc

```diff
@@ -0,0 +1,80 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/termination_criteria/end_walltime.h>
+
+namespace aspect
+{
+  namespace TerminationCriteria
+  {
+    // The start_walltime is made as static and initialized here to 
+    // make sure it is initialized right way as the programe starts.
+    template <int dim>
+    std::time_t
+    EndWalltime<dim>::start_walltime = std::time(NULL);
+
+    template <int dim>
+    bool
+    EndWalltime<dim>::execute()
+    {
+      return( (std::time(NULL)-start_walltime) >= (int)walltime_duration);
+    }
+
+
+    template <int dim>
+    void
+    EndWalltime<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Termination criteria");
+      {
+        prm.declare_entry ("Wall time",
+                           "0",
+                           Patterns::Double (),
```

Comment:
I missed this yesterday: make this `Patterns::Double(0)` to ensure that the user-provided value is not negative.